src/PHPCensor/Worker/BuildWorker.php

```diff
@@ -109,20 +110,30 @@ public function startWorker()
                 $this->pheanstalk->delete($job);
             }
 
-            try {
-                // Logging relevant to this build should be stored
-                // against the build itself.
-                $buildDbLog = new BuildDBLogHandler($build, Logger::INFO);
-                $this->logger->pushHandler($buildDbLog);
+            // Logging relevant to this build should be stored
+            // against the build itself.
+            $buildDbLog = new BuildDBLogHandler($build, Logger::INFO);
+            $this->logger->pushHandler($buildDbLog);
 
+            try {
                 $builder = new Builder($build, $this->logger);
                 $builder->execute();
 
-                // After execution we no longer want to record the information
-                // back to this specific build so the handler should be removed.
-                $this->logger->popHandler();
-                // destructor implicitly call flush
-                unset($buildDbLog);
+            } catch (BuilderException $ex) {
+                $this->logger->addError($ex->getMessage());
+                switch($ex->getCode()) {
+                    case BuilderException::FAIL_START:
```

Comment:
I started 4 builds and all builds hangs in status Build::STATUS_RUNNING with message 'Can`t build - status is not pending' in the middle of build log.